src/views/Course.vue

```diff
@@ -0,0 +1,168 @@
+          accept="application/pdf"
+          @change="onFilePicked"
+        />
+        <div>
+          <v-expansion-panels focusable>
+            <v-expansion-panel v-for="(item, i) in assessments" :key="i">
+              <v-expansion-panel-header>
+                <template>
+                  <v-row no-gutters>
+                    <v-col cols="1" class="assessment-item">
+                      {{ item.grade ? item.grade : 'N/A' }}
+                    </v-col>
+                    <v-col cols="7" class="assessment-item">
+                      {{ item.type }}: {{ item.description }}
+                    </v-col>
+                    <v-col cols="1" class="text--secondary assessment-item">
+                      <v-fade-transition leave-absolute>
+                        <span>
+                          {{ item.weight }}
+                        </span>
+                      </v-fade-transition>
+                    </v-col>
+                    <v-col cols="2" class="text--secondary assessment-item">
+                      <v-fade-transition leave-absolute>
+                        <span>
+                          {{
+                            item.deadline
+                              ? item.deadline
+                              : item.on_going
+                              ? 'On-going'
+                              : 'TBD'
+                          }}
+                        </span>
+                      </v-fade-transition>
+                    </v-col>
+                    <v-col cols="1">
+                      <v-fade-transition leave-absolute>
+                        <span>
+                          <v-btn icon>
+                            <v-icon class="mr-1">
+                              mdi-square-edit-outline
+                            </v-icon>
+                          </v-btn>
+                          <v-btn icon>
+                            <v-icon class="mr-1"> mdi-delete </v-icon>
+                          </v-btn>
+                        </span>
+                      </v-fade-transition>
+                    </v-col>
+                  </v-row>
+                </template>
+              </v-expansion-panel-header>
+              <v-expansion-panel-content>
+                <!-- TODO: Subtasks -->
+              </v-expansion-panel-content>
+            </v-expansion-panel>
+          </v-expansion-panels>
+        </div>
+      </v-sheet>
+    </v-col>
+  </v-row>
+</template>
+
+<script>
+import axios from 'axios';
+
+export default {
+  created() {
+    window.addEventListener('resize', this.handleResize);
+  },
+  data() {
+    return {
+      height: window.innerHeight,
+      file: null,
+      assessments: []
+    };
+  },
+  methods: {
+    handleResize() {
+      this.height = window.innerHeight;
+    },
+    onPickFile() {
+      this.$refs.fileInput.click();
+    },
+    onFilePicked(event) {
+      const { files } = event.target;
+      const filename = files[0].name;
```

Comment:
We should probably add error checking here just in case

Reply:
what error should we catch here btw

Reply:
files is null